aksparse/server/Aksparse.chpl

```diff
@@ -414,13 +423,35 @@ module Aksparse {
         st.addEntry(result2name, new shared SymEntry(result1_uk1.a));
         var result3name = st.nextName();
         st.addEntry(result3name, new shared SymEntry(result3));
+
+        var gb_row_val = getGroupBy(2, [result1name, result3name], ["pdarray", "pdarray"], false, st);
+        var gb_row_val_uklist = get_unique_keys(gb_row_val, st);
+        var gEnt_gb_row_val_uk0: borrowed GenSymEntry = getGenericTypedArrayEntry(gb_row_val_uklist[0], st);
+        var gb_row_val_symentry = toSymEntry(gEnt_gb_row_val_uk0,int);
+        var gb_row_val_uk = getGroupBy(1, [gb_row_val_symentry.name], ["pdarray"], false, st);
+        var gb_row_val_uk_uklist = get_unique_keys(gb_row_val_uk, st);
+        var gEnt_gb_row_val_uk_uk0: borrowed GenSymEntry = getGenericTypedArrayEntry(gb_row_val_uk_uklist[0], st);
+        var gb_row_val_uk_uk_symentry = toSymEntry(gEnt_gb_row_val_uk_uk0,int);
+
+        var len_arr: [0..0] int = result3.size;
+        var segs2 = concatArrays(gb_row_val_uk.segments.a, len_arr); //Fix this?
+        var diffs2 = segs2[1..segs2.size - 1] - segs2[0..segs2.size - 2];
```

Comment:
I think something like this will avoid the having to create the array containing one element and doing a concat
```chapel
ref segs2 = gb_row_val_uk.segments.a;
const sD = segs2.domain;
var diffs2: [sD] int;

forall (i, d, s) in zip(sD, diffs2, segs2) {
    if i == sD.high {
        d = result3.size - s;
    } else {
        d = segs2[i+1] - s;
    }
}
```